### PR TITLE
NINO whitelist, jumbo fix and other improvements

### PR DESCRIPTION
**Pull request by WalshyDev**
## 4.1.13
FlareBot v4.1.13 - Final changes before 4.2

## Added
* Blacklisting! This allows you to blacklist channels and users so they can't use commands! This does **not** block modlogs or anything else! 👍 - **This is being tested in beta**
Command - usage - example
* color - `_color <color>` - `_color red`, `_color #eeefff`, `_color 100,80,20`. The colour command allows for you to see colours and and some values. This is mostly helpful for us but it can also be used for you to test role colours!
* jumbo - `_jumbo <emote>` - `_jumbo :flarekappa:`. Why? Because the world needs bigger emotes!!

## Changed
* NINO can now support multiple messages!
* The usage embed - It should be a bit less "aggressive" looking now.
* In #general of the official guild the bot should now ask users to use #bot-tests

## Fixed
* Hierarchy in mod actions being a tad broken... totally not reversed 👀 

## Beta
* There are cool new settings, you have a _settings command which allows you to customise a lot more how you want the bot to behave! For now you have the option to keep the command the user executes in chat (No more denying FlareBot delete perms and having to deal with purge issues!). You also have the option to blacklist users and channels!!! 

## Random backend stuff
* A bunch of commands got moved to better fit the package names
* Some more @Nullable annotations to ensure we don't make any mistakes in the code.
* More debug information for us to use! Whoo!

### CODE REVIEW

File: src/main/java/stream/flarebot/flarebot/commands/secret/DebugCommand.java [hunk 1]
```diff
@@ -64,6 +64,27 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
...
+
+                    VideoThread.VIDEO_THREADS.activeCount(),
+                    Events.COMMAND_THREADS.activeCount(),
+                    Thread.getAllStackTraces().size(),
+                    MessageUtils.paste(Thread.getAllStackTraces().keySet().stream()
+                            .map(th -> th.getName() + " - " + th.getState() + " (" + th.getThreadGroup().getName() + ")")
+                            .collect(Collectors.joining("\n")))
+            ), channel);
+        } else if (args[0].equalsIgnoreCase("server") || args[0].equalsIgnoreCase("guild")) {
+
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/secret/DebugCommand.java]: These being filled?

File: src/main/java/stream/flarebot/flarebot/commands/secret/DebugCommand.java [hunk 2]
```diff
@@ -64,6 +64,27 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
...
+                    Events.COMMAND_THREADS.activeCount(),
+                    Thread.getAllStackTraces().size(),
+                    MessageUtils.paste(Thread.getAllStackTraces().keySet().stream()
+                            .map(th -> th.getName() + " - " + th.getState() + " (" + th.getThreadGroup().getName() + ")")
+                            .collect(Collectors.joining("\n")))
+            ), channel);
+        } else if (args[0].equalsIgnoreCase("server") || args[0].equalsIgnoreCase("guild")) {
+            
+        } else if (args[0].equalsIgnoreCase("player") || args[0].equalsIgnoreCase("music")) {
+
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/secret/DebugCommand.java]: ^

File: src/main/java/stream/flarebot/flarebot/commands/secret/DebugCommand.java (hunk 1 above)
Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/commands/secret/DebugCommand.java]: Eventually

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 3]
```diff
@@ -402,6 +376,7 @@ protected void run() {
         registerCommand(new UpdateJDACommand());
         registerCommand(new ChangelogCommand());
         registerCommand(new JumboCommand());
+        registerCommand(new ColourCommand());
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: Ok we should probably deside on how we want to hanlde this in code. you know what I'm talking about

File: src/main/java/stream/flarebot/flarebot/commands/random/ColourCommand.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Message message, String[] args, Member member) {
+        if (args.length == 1) {
+            Color colour;
+            if (args[0].matches("^#([A-Fa-f0-9]{6})$")) {
+                colour = Color.decode(args[0]);
+            } else {
+                try {
+                    colour = (Color) Color.class.getField(args[0].toUpperCase()).get(null);
+                } catch (IllegalArgumentException | IllegalAccessException | NoSuchFieldException | SecurityException e) {
+                    MessageUtils.sendErrorMessage("Please specify a correct colour either by a Java colour name or a Hex colour code!", channel);
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/commands/random/ColourCommand.java]: English spelling please

File: src/main/java/stream/flarebot/flarebot/commands/random/JumboCommand.java [hunk 5]
```diff
@@ -56,4 +51,18 @@ public String getUsage() {
     public CommandType getType() {
         return CommandType.RANDOM;
     }
+    
+    private InputStream read(String url, TextChannel channel) {
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/commands/random/JumboCommand.java]: If we can we might want to put a max size on this

File: src/main/java/stream/flarebot/flarebot/util/RandomUtils.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package stream.flarebot.flarebot.util;
+
+import java.util.Collection;
+import java.util.Random;
+
+public class RandomUtils {
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/util/RandomUtils.java]: Do we really use random enough to constitute having a class for random stuff?

Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/util/RandomUtils.java]: Yes to not also deal with random being in the god damn guild JSON, plus it's much quicker

File: src/main/java/stream/flarebot/flarebot/commands/random/JumboCommand.java (hunk 5 above)
Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/commands/random/JumboCommand.java]: What do you mean?

Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/commands/random/JumboCommand.java]: I Imagin file size and image size is bigger for emotes then it is for discord files so I think we'd run into errors if someone has a big emote

Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/commands/random/JumboCommand.java]: Max emote size is 256KB

Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/commands/random/JumboCommand.java]: nvm then

**Pull request merged**